src/cache-service.js

```diff
@@ -0,0 +1,87 @@
+const path = require('path');
+const fs = require('fs');
+const constants = require('./constants');
+
+/**
+ * A File Cache controller for store and retrieve files from disk
+ *
+ * @class CacheService
+ */
+class CacheService {
+    constructor() {
+        this.cachePath = path.join(constants.DATABASE, 'cache');
+    }
+
+    /**
+     * `save` a file on cache folder
+     *
+     * @param {string} fullFilePath
+     * @param {*} data
+     * @returns {promise}
+     * @memberof CacheService
+     */
+    setCache(fullFilePath, data) {
+        return new Promise((resolve, reject) => {
+            try {
+                const file = fs.createWriteStream(path.join(this.cachePath, fullFilePath));
+                file.write(data, (err) => {
+                    if(err) {
+                        throw Error("Can't save file: ", err);
+                    } else {
+                        resolve(true);
+                    }
+                });
+            } catch (err) {
+                reject(err);
+            }
+        });
+    }
+
+    /**
+     * `get` a File from cache folder
+     *
+     * @param {string} fullFilePath
+     * @returns {promise} `Resolve` with file content, `Reject` with false
+     * @memberof CacheService
+     */
+    getCache(fullFilePath) {
+        return new Promise((resolve, reject) => {
+            try {
+                fs.readFile(path.join(this.cachePath, fullFilePath), 'utf8', function (err,data) {
```

Comment:
If I understand this correctly, we're calling the `getCache` method in the following places:
`CacheImageService.getImageFromCache`
`M3uService.buildM3UList`
`M3uService.buildHLSList`

I'd expect these methods (or the methods that consume these methods) to be called potentially numerous times while a user is managing their dizquetv channels. If I'm mistaken, let me know.

With the implementation here, we'd be hitting the file system every single time those functions are called.

We may be able to gain performance speed by string the cached values in memory - in a JS object here - whenever we want to retrieve the cache.

That may look like:
1) On dizquetv start, preload commonly accessed cache into memory (in this service)
2) On `getCache` being called, return the variable from memory rather than reading from a file every time
3) Whenever we `setCache`, update the variable stored in memory _and_ update the file storage.

The negative here being memory impact. I'm not certain how large we can expect the files to be, and it may or may not be large enough that storing in memory uses up a painfully high amount of resources. Might be worth weighing one versus the other if we haven't already, though.